### PR TITLE
Slight Override Refactor + Better Docs

### PR DESCRIPTION
**Pull request by rambleraptor**
I made some override refactors as I tried to handle Ansible backporting. I'm going a different route for the backporting, but I think the refactors are still useful.

This adds in some documentation explaining what exactly is happening with the overrides and hopefully helps make the code more self-documenting.

This PR is a no-op and passes all tests. The overrides still work in the same fundamental manner.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

**Comment by rambleraptor** [comment]
@SirGitsalot - normal code review
@rileykarson - optional, but I feel like you'd have feelings on this.

File: overrides/runner.rb [hunk 1]
```diff
@@ -23,76 +23,122 @@ module Overrides
   # description:
   #
   # overrides: !ruby/object:Overrides::ResourceOverrides
+  #   product: # Allows us to override information about the product.
+  #     prefix: magicproduct
```
Comment (rileykarson) [inline overrides/runner.rb]: Is this an actually override we have / have we overriden the product for a resource yet? I can see it being a useful feature, I'm just not familiar with any occurences. Also I'd lean towards `legacy_name` over `prefix`, similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/master/products/cloudbuild/terraform.yaml#L15

File: overrides/runner.rb [hunk 2]
```diff
@@ -23,76 +23,122 @@ module Overrides
...
   class Runner
-    class << self
-      def initialize(api, overrides,
-                     res_override_class = Overrides::ResourceOverride,
-                     prop_override_class = Overrides::PropertyOverride)
-        @api = api
-        @overrides = overrides
-        @res_override_class = res_override_class
-        @prop_override_class = prop_override_class
+    # Internal information for applying overrides.
```
Comment (rileykarson) [inline overrides/runner.rb]: This doc string basically just repeats the name of the class; what kind of internal information, and how is it useful?

File: overrides/runner.rb [hunk 3]
```diff
@@ -23,76 +23,122 @@ module Overrides
...
       # return a new Api::Resource with overrides applied.
-      def build_resource(old_resource, res_override, override_classes)
-        res_override = override_classes[:resource].new if res_override.nil? || res_override.empty?
+      def build_resource(old_resource, overrides)
+        # Set up the ResourceOverride object.
+        res_override = overrides.resource_overrides(old_resource.name)
         res_override.validate
         res_override.apply old_resource
 
+        # Create new Api::Resource + apply all provider-specific values.
```
Comment (rileykarson) [inline overrides/runner.rb]: This comment makes me think that `set_additional_values` wants to be named a little differently potentially? Assuming both call sites are doing the same thing.

File: overrides/runner.rb [hunk 4]
```diff
@@ -103,27 +149,32 @@ def build_resource(old_resource, res_override, override_classes)
...
 
-      # Given a Api::Type property and a hash of properties, create a new Api::Type property
-      # This will handle NestedObjects, Arrays of NestedObjects of arbitrary length
-      def build_property(old_property, property_overrides, override_classes, prefix = '')
-        property_overrides = {} if property_overrides.nil?
-        new_prop = build_primitive_property(old_property,
-                                            property_overrides["#{prefix}#{old_property.name}"],
-                                            override_classes)
+      # Given a old_property and our overrides, build a new Api::Type object
+      # that's has all values overriden.
```
Comment (SirGitsalot) [inline overrides/runner.rb]:
```suggestion
      # that has all values overriden.
```

**Comment by rileykarson** [comment]
Should we go for a merge or close this as outdated?

**Comment by rileykarson** [comment]
Spoke with @rambleraptor over chat- closing as outdated.

**Pull request closed without merging**